uwblueprint/childrens-aid-society
# Sohail/intake state population

## Notion ticket link

[Implement Intake state populating](https://www.notion.so/uwblueprintexecs/Implement-Intake-state-populating-eb47e1c789b9484a9aa5c16b6a0dd957?pvs=4)



## Implementation description
* Built on top of changes from https://github.com/uwblueprint/childrens-aid-society/pull/184
* populate review page with mock intake data, when lead to review case details page from review button
* move default/initial intake values into new constants/DefaultIntakeValues.ts file, due to repeated use of default values
* expand IntakeValueContext.ts by declaring referral, court and program detail states here (previously declared in IntakePage.tsx)


## Steps to test
1. Click on Test Status Modal > "Review" button - should be redirected to intake page, with review case details step open 
![image](https://github.com/uwblueprint/childrens-aid-society/assets/44615627/c53a2968-0871-40f0-856a-3b4f2079b962)
![image](https://github.com/uwblueprint/childrens-aid-society/assets/44615627/854fc860-aa4d-4f70-a71b-22ccea715ee3)
![image](https://github.com/uwblueprint/childrens-aid-society/assets/44615627/7efe8381-c643-4cc2-b6ba-8870f692db6a)
![image](https://github.com/uwblueprint/childrens-aid-society/assets/44615627/a5fa5cc3-6606-4897-9986-52653fb47a42)
![image](https://github.com/uwblueprint/childrens-aid-society/assets/44615627/9a54b46f-0d29-4a27-9988-8ea90330ff55)

2. At review page details page mock referral, court and program details should be displayed 
3. Complete intake steps and submit case, should go through as normal
4. If you return to dashboard and create a new case, should start at first step and proceed as expected (mock information should not be present now, it should be unfilled)


## What should reviewers focus on?
* To see if logic of intake steps breaks (if certain steps are skipped, or if it doesn't start at the first step when creating a new case or going straight to '/intake')
* To look out for if any mock information persists when it shouldn't be there (starting a new intake form)

## Checklist
